src/simplification/SimplifyExtend.ts

```diff
@@ -0,0 +1,30 @@
+import { CommonModel } from "models";
+import { Schema } from "models/Schema";
+import Simplifier from "./Simplifier";
+type output = {newModels: CommonModel[] | undefined; extendingSchemas: string[] | undefined};
+
+/**
+ * Find out which common models we should extend
+ * 
+ * @param schema to find extends of
+ */
+export default function simplifyExtend(schema: Schema | boolean, simplifier : Simplifier) : output {
+  let models : CommonModel[] | undefined;
+  let extendingSchemas : string[] | undefined;
```

Comment:
You can assign empty array to those properties and then change code from this:

```ts
let models : CommonModel[] | undefined;
let extendingSchemas : string[] | undefined;

extendingSchemas = [...(extendingSchemas || []), rootSimplifiedModel.$id];
models = [...(models || []), ...simplifiedModels];
```

to 

```ts
let models : CommonModel[] = [];
let extendingSchemas : string[] = [];

extendingSchemas = [...extendingSchemas, rootSimplifiedModel.$id];
models = [...models), ...simplifiedModels];
```